libs/sdk-core/src/invoice.rs

```diff
@@ -172,7 +172,7 @@ pub fn parse_invoice(bolt11: &str) -> Result<LNInvoice> {
     let converted_hints = invoice_hints.iter().map(RouteHint::from_ldk_hint).collect();
     // return the parsed invoice
     let ln_invoice = LNInvoice {
-        bolt11: bolt11.to_string(),
+        bolt11: invoice.to_string(),
```

Comment:
The resulting invoice string can be slightly different from the actual bolt11 invoice string passed as input by parsing and re-encoding.
I think it's best to do something like:

```rust
let bolt11 = bolt11
        .strip_prefix("lightning:")
        .unwrap_or(bolt11);
let signed = bolt11.parse::<SignedRawInvoice>()?;
```

And pass `bolt11` here.